src/dune_pkg/fetch.mli

```diff
@@ -19,6 +19,33 @@ val fetch
   -> OpamUrl.t
   -> (unit, failure) result Fiber.t
 
+module Git_repo : sig
+  type t
+  type rev
+
+  module Remote : sig
+    type t
+
+    val update : t -> unit Fiber.t
+    val default_branch : t -> string option Fiber.t
+
+    module At_rev : sig
+      type t
+
+      val content : t -> Path.t -> string option Fiber.t
```

Comment:
Perhaps the signature should be changed to `Path.Local.t`? To make sure that we only provide paths inside the repository

Reply:
`Path.Local.t` is a relative path that doesn't start with `..`. Exactly what we need here. I've modified the code to use it.